### PR TITLE
[kai/child-routes] Children service delete

### PR DESCRIPTION
**Pull request by kai-wei-mo**
## Notion ticket link

N/A

## Implementation description
N/A

## Steps to test
* Visual inspection
* Unit tests

## What should reviewers focus on?
* N/A

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/tests/functional/test_child_service.py [hunk 1]
```diff
@@ -146,3 +169,13 @@ def test_missing_field(child_service):
...
     with pytest.raises(Exception):
         child_service.add_new_child(param)
+
+
+def test_delete_success(child_service):
+    child_service.delete_child(1)
+    assert Child.query.get(1) is None
+
+
+def test_delete_fail(child_service):
```
Comment (jennchenn) [inline backend/python/tests/functional/test_child_service.py:179]: nit

```suggestion
def test_delete_invalid_id_fail(child_service):
```

File: backend/python/app/resources/child_dto.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def validate(self):
             error_list.append("The first_name supplied is invalid")
         if not self.last_name or not type(self.last_name) == str:
             error_list.append("The last_name supplied is invalid")
-        if self.date_of_birth and not isinstance(self.date_of_birth, datetime.date):
+        if self.date_of_birth and not type(self.date_of_birth) in [str, datetime.date]:
```
Comment (jennchenn) [inline backend/python/app/resources/child_dto.py:31]: is date validation happening elsewhere if we are accepting a string here?

Comment (kai-wei-mo) [reply backend/python/app/resources/child_dto.py]: added

File: backend/python/app/resources/child_dto.py [hunk 3]
```diff
@@ -50,4 +44,18 @@ def validate(self):
                 "has_kinship_provider and has_foster_placement cannot have the same boolean value"
             )
 
+        # optional args
+        if self.intake_id and not type(self.intake_id) == int:
+            error_list.append("The intake_id supplied is invalid")
+        if self.date_of_birth:
+            if type(self.date_of_birth) == str:
```
Comment (jennchenn) [inline backend/python/app/resources/child_dto.py:51]: just trying to understand - when do you expect this to be a string vs. an actual datetime?

**Review by jennchenn — APPROVED** [review]
left a question but can be merged anyway to unblock

**Pull request merged**